src/azure-cli/azure/cli/command_modules/databoxedge/generated/_params.py

```diff
@@ -0,0 +1,253 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    get_enum_type,
+    resource_group_name_type,
+    get_location_type
+)
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+from ..action import AddSku
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('databoxedge device list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('expand', type=str, help='Specify $expand=details to populate additional fields related to the '
+                   'resource or Specify $skipToken=<token> to populate the next page in the list.')
+
+    with self.argument_context('databoxedge device show') as c:
+        c.argument('device_name', options_list=['--name', '-n', '--device-name'], type=str, help='The device name.',
+                   id_part='name')
+        c.argument('resource_group_name', resource_group_name_type)
+
+    with self.argument_context('databoxedge device create') as c:
+        c.argument('device_name', options_list=['--name', '-n', '--device-name'], type=str, help='The device name.')
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('location', arg_type=get_location_type(self.cli_ctx), required=False,
+                   validator=get_default_location_from_resource_group)
+        c.argument('tags', tags_type)
+        c.argument('sku', action=AddSku, nargs='+', help='The SKU type.')
+        c.argument('etag', type=str, help='The etag for the devices.')
+        c.argument('data_box_edge_device_status', options_list=['--status'], arg_type=get_enum_type(['ReadyToSetup',
```

Comment:
could we get the enum type from sdk

Reply:
I have asked @qiaozha for support, guess developing?

Reply:
If there is enum definition in SDK, you could rewrite it now.
Getting from SDK could adjust with api version update.

Reply:
done